executor/write.go

```diff
@@ -1375,6 +1394,45 @@ func getUpdateColumns(assignList []*expression.Assignment, schemaLen int) ([]boo
 	return assignFlag, nil
 }
 
+func (e *UpdateExec) fetchChunkRows(goCtx goctx.Context) error {
+	fields := e.children[0].Schema().GetTypes()
+	for {
+		chk := chunk.NewChunk(fields)
+		err := e.children[0].NextChunk(goCtx, chk)
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		if chk.NumRows() == 0 {
+			break
+		}
+
+		for rowIdx := 0; rowIdx < chk.NumRows(); rowIdx++ {
```

Comment:
why not just evaluate the expressions in `e.OrderedList` based on the chunk ?

Reply:
the current `NextChunk` for `UpdateExec` is less efficient than `Next`, I think maybe we need another mechanism to handle chunk execution instead of reusing the `exec` function.

Reply:
this is not a trival work, we should change this step by step.